pkg/fanal/types/artifact.go

```diff
@@ -156,8 +157,16 @@ func (id *PkgIdentifier) Empty() bool {
 }
 
 func (id *PkgIdentifier) Match(s string) bool {
+	// `s` can be an unescaped string
+	// e.g. CycloneDX unescapes BOM Links - https://github.com/CycloneDX/cyclonedx-go/blob/b9654ae9b4705645152d20eb9872b5f3d73eac49/link.go#L136
+	// So we need to match escaped and unescaped BOMRef
+	var unescapedBomRef string
+	if ref, err := url.QueryUnescape(id.BOMRef); err == nil {
```

Comment:
I also thought about an unshielded PURL, but that [Match function](https://github.com/aquasecurity/trivy/blob/b1c59d89722a6f978d75f6cc9db466daaff55130/pkg/fanal/types/artifact.go#L159) is only used for the CycloneDX Vex.
CycloneDX Vex is used only in `sbom` mode. This will fill in the BomRef field.
I think we can start with changes just for BomRef.